pilot/pkg/status/resourcelock.go

```diff
@@ -262,7 +263,7 @@ type IstioGenerationProvider struct {
 }
 
 func (i *IstioGenerationProvider) SetInner(c any) {
-	panic("not supported for this type")
+	panic("not supported for this type: " + reflect.TypeOf(c).String())
```

Comment:
The panic is because they called SetInner using a `*IstioGenerationProvider` receiver not because of the type of the thing passed in. It's probably not necessary to reflect the type of what they passed in.

Reply:
So, tbc when you say `It's probably not necessary to reflect the type of what they passed in.` does that mean if we create another type in the future (similar to the ServiceEntry situation)... would the "type that they pass in" be that type and the analyzer would attempt to write status for that type?

If that's the case (which was my assumption when adding this here) then knowing the reason for panicking, a.k.a this specific type is not supported, would be helpful IMO. For the user but most definitely for the developer fixing this.

But if this assumption is wrong, then I agree. Doesn't make much sense to get the type of GP. It's right there in the code, seems redundant to include it in the panic. But if the assumption is correct, I think having extra data about the incorrect type being passed in and not handled (aka, panicking and saying this type has no support without saying what type) is definitely useful.

Reply:
You're getting a panic because of the receiver not because of what you passed in. It actually doesn't matter what you pass in it will always be unsupported because the receiver type doesn't actually support this but is "implementing" it in order to implement an interface. Even if you passed in a `NopStatusManipulator`, which is a type that does fully implement this, you will get the panic because the input type isn't relevant to the behavior.

Reply:
Ah ok, then yeah this isn't helpful. Good point.